### PR TITLE
feat(core/pane-layout): provide close/open event and default slot

### LINKED ISSUE #1512: Add close event to ix-pane
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/siemens/ix/blob/main/CONTRIBUTING.md).
- [X] I have not leaked any internal/restricted information like screenshots, videos, code snippets, links etc.

### Suggestion / feature request

I suggest to add a click event binding when the 'x' or '>>' icons are clicked to close the currently expanded pane, e.g.

```js
// angular template
<ix-pane-layout [variant]="variant" [layout]="layout" [borderless]="borderless">
  <ix-pane (close)="paneClosed()" heading="Pane Right" slot="right" size="33%">
    <p>This is the right pane.</p>
  </ix-pane>

  <div slot="content">
    // some content
  </div>
</ix-pane-layout>

// in the component code
paneClosed() {
  // stuff to do on pane closed, e.g. change url or deselect current row
}
```

This would allow to perform something right after the close action. In the current version there is only the `expandedChanged` event. It's event data provides not the sufficient details to decide whether the pane was closed by the pane close icon or not.

This are the actual icons: 

image
image

### PR DESCRIPTION
**Pull request by matthiashader**
## 💡 What is the current behavior?

GitHub Issue Number: Closes #1512, [IX-1829]
- user don't get notified, before the user expands the pane via the icon button
- users have to name the slot in the pane-layout in order to utilize it

## 🆕 What is the new behavior?

- Close event provided, with the option to prevent the event (prevent from opening / closing)
- besides the <slot name="content"> a default fallback <slot> have been provided.
- Note: A better name for the event could be provided
## 🏁 Checklist

A pull request can only be merged if all of these conditions are met (where applicable):

- [ ] 🦮 Accessibility (a11y) features were implemented
- [ ] 🗺️ Internationalization (i18n) - no hard coded strings
- [ ] 📲 Responsiveness - components handle viewport changes and content overflow gracefully
- [ ] 📄 Documentation was reviewed/updated (`pnpm run docs`)
- [x] 🧪 Unit tests were added/updated and pass (`pnpm test`)
- [ ] 📸 Visual regression tests were added/updated and pass ([Guide](https://github.com/siemens/ix/blob/main/CONTRIBUTING.md#visual-regression-testing))
- [x] 🧐 Static code analysis passes (`pnpm lint`)
- [x] 🏗️ Successful compilation (`pnpm build`, changes pushed)

### CODE REVIEW

File: .changeset/seven-coins-fly.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@siemens/ix-angular': patch
+'@siemens/ix': patch
+'@siemens/ix-vue': patch
```
Comment (nuke-ellington) [inline .changeset/seven-coins-fly.md:4]:
```suggestion
'@siemens/ix-angular': minor
'@siemens/ix': minor
'@siemens/ix-vue': minor
```

**Review by nuke-ellington — CHANGES_REQUESTED** [review]
Why not provide dedicated expand and collapse events?

File: .changeset/seven-coins-fly.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+'@siemens/ix-angular': minor
+'@siemens/ix': minor
+'@siemens/ix-vue': minor
+---
+
+feat(core/pane-layout): provide close/open event and default slot #1563
```
Comment (jul-lam) [inline .changeset/seven-coins-fly.md:7]: Adapt changeset to new format

**Review by jul-lam — CHANGES_REQUESTED** [review]

File: .changeset/seven-coins-fly.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+'@siemens/ix-angular': minor
+'@siemens/ix': minor
+'@siemens/ix-vue': minor
+---
+
+Update expandedChange event to trigger only on user interactions and refine default slot behavior in ix-pane-layout.
```
Comment (nuke-ellington) [inline .changeset/seven-coins-fly.md:7]:
```suggestion
Update expandedChange event to trigger only on user interactions and add unnamed default slot for ix-pane-layout content.
```

**Pull request merged**